src/databricks/sql/client.py

```diff
@@ -1119,7 +1161,7 @@ def __init__(
         self._arrow_schema_bytes = execute_response.arrow_schema_bytes
         self._next_row_index = 0
 
-        if execute_response.arrow_queue:
+        if execute_response.arrow_queue or async_op:
```

Comment:
It does not depend on arrow, currently in our codebase all data is named as arrow_queue be it arrow queue or column queue. So in this statement I am checking if data is already present or if it is an async operation don't do anything.

Reply:
My point is `ResultSet` should not couple with `async_op` as I do not see it has something related to async. If you want to force to use arrow_queue, please use `force_arrow_queue` or similar parameter instead of `async_op` in the constructor.

Reply:
Fixed this